### PR TITLE
Document that `farcall` returns `c` in `a` for some code

### LINKED ISSUE #915: Comment that some code depends on `farcall` returning the value of `c` in `a`
See https://discord.com/channels/442462691542695948/487424856913346580/971165772632186980:

> `TryTileCollisionEvent` loads `wFacingTileID` into `c`, then `farcall`s `CheckFacingTileForStdScript`, which preserves `c`, but since it's a `farcall`, implicitly copies `c` back into `a` on return.  Then it calls `CheckCutTreeTile` which checks this value.  Notice how every other <code>Check<i>&lt;x&gt;</i>Tile</code> call in the same function explicitly loads `wFacingTileID` again.

### PR DESCRIPTION
**Pull request by coltongit**
Fixes #915
Other code may also do the same thing, would be worth a fine-toothed comb through to check again

### CODE REVIEW

**Comment by mid-kid** [comment]
That's a huge ass comment for something that could be commented as "expects c to be copied into a".

File: engine/overworld/events.asm [hunk 1]
```diff
@@ -1081,6 +1081,7 @@ LoadScriptBDE::
 	scf
 	ret
 
+; expects c to be copied into a
```
Comment (vulcandth) [inline engine/overworld/events.asm:1084]: Please move this comment above `call CheckCutTreeTile` like this:

```diff
-; expects c to be copied into a
 TryTileCollisionEvent::
 	call GetFacingTileCoord
 	ld [wFacingTileID], a
 	ld c, a
 	farcall CheckFacingTileForStdScript
 	jr c, .done

+	; CheckCutTreeTile expects c to be copied into a by farcall
 
 	call CheckCutTreeTile
 	jr nz, .whirlpool
 	farcall TryCutOW
 	jr .done
```

File: engine/overworld/events.asm [hunk 2]
```diff
@@ -1087,7 +1087,8 @@ TryTileCollisionEvent::
 	ld c, a
 	farcall CheckFacingTileForStdScript
 	jr c, .done
-
+	
+	; CheckCutTreeTile expects c to be copied into a by farcall
 	call CheckCutTreeTile
```
[the remark this answers is not in the document]
Comment (Rangi42) [reply engine/overworld/events.asm:1092]: That might be a correction to make to STYLE.md instead. In practice we do generally keep comments "attached" to the line or "paragraph" they're explaining:

```
	; This call is unusual somehow.
	call SomeFunction
```

```
	; Call the function at de
	ld a, [de]
	ld l, a
	inc de
	ld a, [de]
	ld h, a
	ld de, .return
	push de
	jp hl

.return
```

[the remark this answers is not in the document]
Comment (vulcandth) [reply engine/overworld/events.asm:1092]: I support making the change in STYLE.md then. As I actually prefer keeping them attached personally.

[the remark this answers is not in the document]
Comment (coltongit) [reply engine/overworld/events.asm:1092]: Yeah, this is something for STYLE.md

**Review by vulcandth — APPROVED** [review]
Alright. Looks good to me! Thanks :)

File: engine/overworld/events.asm [hunk 3]
```diff
@@ -1087,7 +1087,8 @@ TryTileCollisionEvent::
 	ld c, a
 	farcall CheckFacingTileForStdScript
 	jr c, .done
-
+
```
Comment (Rangi42) [inline engine/overworld/events.asm:1090]:
```suggestion

```

File: engine/overworld/events.asm [hunk 4]
```diff
@@ -1087,7 +1087,7 @@ TryTileCollisionEvent::
 	ld c, a
 	farcall CheckFacingTileForStdScript
 	jr c, .done
-
+	; CheckCutTreeTile expects c to be copied into a by farcall
```
Comment (Rangi42) [inline engine/overworld/events.asm:1090]: I just meant, remove the trailing space.
```suggestion

	; CheckCutTreeTile expects c to be copied into a by farcall
```

File: engine/overworld/events.asm [hunk 5]
```diff
@@ -1088,6 +1088,7 @@ TryTileCollisionEvent::
 	farcall CheckFacingTileForStdScript
```
Comment (Rangi42) [inline engine/overworld/events.asm:1088]:
```suggestion
	; CheckFacingTileForStdScript preserves c, and
	; farcall copies c back into a.
	farcall CheckFacingTileForStdScript
```

File: engine/overworld/events.asm [hunk 6]
```diff
@@ -1088,6 +1088,7 @@ TryTileCollisionEvent::
...
 	jr c, .done
 
+	; CheckCutTreeTile expects c to be copied into a by farcall
```
Comment (Rangi42) [inline engine/overworld/events.asm:1091]:
```suggestion
	; CheckCutTreeTile expects a == [wFacingTileID], which
	; it still is after the previous farcall.
```

**Review by Rangi42 — APPROVED** [review]
This will also need doing for pokegold.

**Comment by vulcandth** [comment]
Created the pokegold PR.

**Pull request merged**